File: Adv.API/ClientApp/src/app/models/searchFlatCriteria.ts
export class SearchFlatCriteria{
    rooms: number;
    city: number;
    priceMin: number;
    priceMax: number;
    rentType: number;
    pageNumber: number;
    public constructor(init?: Partial<SearchFlatCriteria>){
        Object.assign(this, init);
    }
}

File: Adv.API/Models/Criteria/SearchFlatCriteria.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Adv.API.Models.Criteria
{
    public class SearchFlatCriteria
    {
        public byte City { get; set; }
        public byte Rooms { get; set; }
        public decimal PriceMin { get; set; }
        public decimal PriceMax { get; set; }
        public byte RentType { get; set; }

    }
}
